llvm/include/llvm/CAS/ObjectStore.h

```diff
@@ -156,7 +158,16 @@ class ObjectStore {
   /// Load the object referenced by \p Ref.
   ///
   /// Errors if the object cannot be loaded.
-  virtual Expected<ObjectHandle> load(ObjectRef Ref) = 0;
+  /// \returns \c std::nullopt if the object is missing from the CAS.
+  virtual Expected<std::optional<ObjectHandle>> load(ObjectRef Ref) = 0;
+
+  /// Asynchronous version of \c load.
+  virtual void loadAsync(
+      ObjectRef Ref,
+      unique_function<void(Expected<std::optional<ObjectHandle>>)> Callback);
+
+  /// Like \c load but returns an error if the object is missing.
+  Expected<ObjectHandle> loadExisting(ObjectRef Ref);
```

Comment:
I prefer the names you used for `getProxy` below; could we use the same idea for `load`?
* load -> loadIfExists
* loadAsync -> loadIfExistsAsync
* loadExisting -> load

Reply:
Done: https://github.com/apple/llvm-project/pull/6458/commits/169516238f4282952bef7b7408e802d03354a031